sfs/mono/source.py

```diff
@@ -705,6 +705,80 @@ def plane_averaged_intensity(omega, x0, n0, grid, c=None):
     return util.XyzComponents([i * n for n in n0])
 
 
+def pulsating_sphere(omega, x0, radius, amplitude, grid, c=None):
+    """Sound pressure of a pulsating sphere.
+
+    Parameters
+    ---------
+    omega : float
+        Frequency of pulsating sphere
+    x0 : (3,) array_like
+        Center of sphere.
+    radius : float
+        Radius of sphere.
+    amplitude : float
+        Amplitude of displacement.
+    grid : triple of array_like
+        The grid that is used for the sound field calculations.
+        See `sfs.util.xyz_grid()`.
+    c : float, optional
+        Speed of sound.
+
+    Returns
+    -------
+    numpy.ndarray
+        Sound pressure at positions given by *grid*.
+    """
+    if c is None:
+        c = defs.c
+    k = util.wavenumber(omega, c)
+    x0 = util.asarray_1d(x0)
+    grid = util.as_xyz_components(grid)
+
+    r = np.linalg.norm(grid - x0)
+    r[r <= radius] = np.nan
+    theta = np.arctan(1, k * r)
+    impedence = defs.rho0 * c * np.cos(theta) * np.exp(1j * theta)
+    return impedence * 1j * omega * amplitude * radius / r \
+        * np.exp(-1j * k * (r - radius))
```

Comment:
I'm getting the warning

    /home/mg/git/sfs-python/sfs/mono/source.py:743: RuntimeWarning: invalid value encountered in true_divide
      * np.exp(-1j * k * (r - radius))

I guess this is because of the NaN values you are using.

Is it possible to add the NaN values after the calculation?
Or excluding them from the calculation somehow?

You could also silence the warning, but this should only be the last resort if nothing else works.

Reply:
Yes, thanks, the warning is gone now!

I don't think I did anything to switch this warning on. At least I cannot remember.

I'm using NumPy version `1.16.0rc1` from a Debian package.